File: Online/NSImage+SY.m
//
//  NSImage+SY.m
//  Online
//
//  Created by Stan Chevallier on 19/11/2015.
//  Copyright © 2015 Syan. All rights reserved.
//

#import "NSImage+SY.h"

@implementation NSImage (SY)

+ (NSImage *)imageWithColor:(NSColor *)color
{
    return [self imageWithColor:color size:NSMakeSize(1, 1)];
}

+ (NSImage *)imageWithColor:(NSColor *)color size:(NSSize)size
{
    NSImage *image = [[NSImage alloc] initWithSize:size];
    [image lockFocus];
    [color drawSwatchInRect:(NSRect){NSZeroPoint, size}];
    [image unlockFocus];
    return image;
}

@end

File: Online/Extensions/NSImage+SY.swift
//
//  NSImage+SY.swift
//  Online
//
//  Created by syan on 13/04/2024.
//  Copyright © 2024 Syan. All rights reserved.
//

import Cocoa

extension NSImage {
    convenience init(color: NSColor, size: NSSize = .init(width: 1, height: 1)) {
        self.init(size: size)
        lockFocus()
        color.drawSwatch(in: NSRect(origin: .zero, size: size))
        unlockFocus()
    }
}
